src/Components/Blazor/Blazor/src/Hosting/EntrypointInvoker.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Blazor.Hosting
+{
+    internal static class EntrypointInvoker
+    {
+        public static Task InvokeEntrypointAsync(IntPtr entrypointMethodHandleValue, string[] args)
```

Comment:
Do you think it's easy to write a unit test for this?

Reply:
I'm going to bet the answer is no. We need to have a **pointer** to a method. I'm sure that's not going to behave the same way on CoreCLR and Mono-WASM